global: generalize color classes and variables

jennur opened the pull request:
Closes https://github.com/inveniosoftware/invenio-app-rdm/issues/1503

Needs https://github.com/inveniosoftware/invenio-theme/pull/295

jrcastro2 commented on invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/requests.js:
```diff
@@ -466,35 +466,35 @@ const RequestCancelButtonMobileWithConfig = parametrize(RequestCancelButton, {
 
 const CommunitySubmission = () => (
-  <LabelTypeSubmission className="rel-mr-1" size="small" color="blue" />
+  <LabelTypeSubmission className="rel-mr-1" size="small" className="primary" />
```
```suggestion
  <LabelTypeSubmission className="rel-mr-1 primary" size="small" />
```

jrcastro2 commented on invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/requests.js:
```diff
@@ -466,35 +466,35 @@ const RequestCancelButtonMobileWithConfig = parametrize(RequestCancelButton, {
 
 const CommunityInvitation = () => (
-  <LabelTypeInvitation className="rel-mr-1" size="small" color="blue" />
+  <LabelTypeInvitation className="rel-mr-1" size="small" className="primary" />
```
```suggestion
  <LabelTypeInvitation className="rel-mr-1 primary" size="small" />
```

jrcastro2 commented on invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/requests.js:
```diff
@@ -466,35 +466,35 @@ const RequestCancelButtonMobileWithConfig = parametrize(RequestCancelButton, {
 
 const Submitted = () => (
-  <LabelStatusSubmit className="rel-mr-1" size="small" color="blue" />
+  <LabelStatusSubmit className="rel-mr-1" size="small" className="primary" />
```
```suggestion
  <LabelStatusSubmit className="rel-mr-1 primary" size="small" />
```

jrcastro2 commented on invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/requests.js:
```diff
@@ -466,35 +466,35 @@ const RequestCancelButtonMobileWithConfig = parametrize(RequestCancelButton, {
 
 const Deleted = () => (
-  <LabelStatusDelete className="rel-mr-1" size="small" color="red" />
+  <LabelStatusDelete className="rel-mr-1" size="small" className="negative" />
```
```suggestion
  <LabelStatusDelete className="rel-mr-1 negative" size="small" />
```

jrcastro2 commented on invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/requests.js:
```diff
@@ -466,35 +466,35 @@ const RequestCancelButtonMobileWithConfig = parametrize(RequestCancelButton, {
 
 const Accepted = () => (
-  <LabelStatusAccept className="rel-mr-1" size="small" color="green" />
+  <LabelStatusAccept className="rel-mr-1" size="small" className="positive" />
```
```suggestion
  <LabelStatusAccept className="rel-mr-1 positive" size="small" />
```

jrcastro2 commented on invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/requests.js:
```diff
@@ -466,35 +466,35 @@ const RequestCancelButtonMobileWithConfig = parametrize(RequestCancelButton, {
 
 const Declined = () => (
-  <LabelStatusDecline className="rel-mr-1" size="small" color="red" />
+  <LabelStatusDecline className="rel-mr-1" size="small" className="negative" />
```
```suggestion
  <LabelStatusDecline className="rel-mr-1 negative" size="small" />
```

jrcastro2 commented on invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/requests.js:
```diff
@@ -466,35 +466,35 @@ const RequestCancelButtonMobileWithConfig = parametrize(RequestCancelButton, {
 
 const Cancelled = () => (
-  <LabelStatusCancel className="rel-mr-1" size="small" color="grey" />
+  <LabelStatusCancel className="rel-mr-1" size="small" className="neutral" />
```
```suggestion
  <LabelStatusCancel className="rel-mr-1 neutral" size="small" />
```

jrcastro2 commented on invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/requests.js:
```diff
@@ -466,35 +466,35 @@ const RequestCancelButtonMobileWithConfig = parametrize(RequestCancelButton, {
 
 const Expired = () => (
-  <LabelStatusExpire className="rel-mr-1" size="small" color="grey" />
+  <LabelStatusExpire className="rel-mr-1" size="small" className="neutral" />
```
```suggestion
  <LabelStatusExpire className="rel-mr-1 neutral" size="small" />
```

jrcastro2 commented on invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/collections/grid.variables:
```diff
@@ -1,4 +1,4 @@
          Invenio App RDM Grid Variables
 ***********************************************/
-@gridLightGreyBackgroundColor: #F8F8F9;
+@gridNeutralBackgroundColor: #F8F8F9;
```
Question: If it's called neutral, shouldn't this just use @neutralColor instead of a concrete value? Just asking for consistency in the naming, if it's not the same value than maybe rename it? WDYT?

jennur replied:
This is the light background color, which is brighter than the `@neutralColor` which is used on labels etc, so it shouldn't be the same, but maybe this one should be added to invenio-theme as well... ping @kpsherva

jennur replied:
Ah sorry, now I see what you mean. You're right, maybe it should rather be `@gridLightNeutralBackgroundColor` or just `@lightNeutralColor` or something similar.

jennur commented:
## Updates

- Moved `.ui.grid`-color classes and variables to `invenio-theme`
- Fixed some linting errors/warnings (indentation, array-indices used in `key`)
